### DISCUSSION TITLE
Potential memory leak in PaddleOCR?

### DISCUSSION THREAD

**Discussion by nikos-livathinos** [PaddlePaddle/PaddleOCR#7967] [Q&A]
I have noticed some weird memory usage when evaluating the performance of PaddleOCR:

- When PaddleOCR processes new images of a sequence, there is a **constant increase in memory usage** of the process.
- The time profile of the memory usage has sudden steps to higher memory levels.
- If the system sets an upper bound in the memory consumption, the paddle process is eventually killed. It won't give allocated memory back, ever.
- This behaviour can be observed both with the C++ and the Python interface.

Eventually **it is impossible to keep running a PaddleOCR process as a service because the system runs out of memory and the process is killed**.

Can you provide insights on this memory usage pattern? Do you have any remedy?

The following issue describes the tests in detail: https://github.com/PaddlePaddle/PaddleOCR/issues/7823

**Comment by LDOUBLEV** [comment]
Thanks for your feedback.

Next, I will focus on analyzing the problem of memory leaks.

And you can join the discussion in our slack group.

https://join.slack.com/t/paddleocr/shared_invite/zt-1hp6bcg4x-YDQZhRMhENMUT7iMS7~EIA

**Reply by ammar-baga** [reply]
> Thanks for your feedback.
> 
> Next, I will focus on analyzing the problem of memory leaks.
> 
> And you can join the discussion in our slack group.
> 
> https://join.slack.com/t/paddleocr/shared_invite/zt-1hp6bcg4x-YDQZhRMhENMUT7iMS7~EIA

Hey @LDOUBLEV any updates on this memory leak issue? or any workaround? for me even reloading the model doesn't free up all the memory it occupied

**Comment by MuhammedOzdogan** [comment]
Do you have any update?
I also see similar issue, when I test with small images it succeeds however if I try with bigger image and get an Out Of Memory error I can't even use the small image that succeed previously all other OCR operations after the Out Of Memory error fails.
